lib/generator.ts

```diff
@@ -210,20 +211,51 @@ export class Generator extends Cli {
     return this.getArgument(ECliArgument.output) as string
   }
 
+  /**
+   * Append this template where {0} is replaced with the name/path of the entry module.
+   */
+  private getTemplate(): string | undefined {
+    return this.getArgument(ECliArgument.template) as string | undefined
+  }
+
   /**
    * Checks if script is forced to use its built-in TSC
    */
   private useTestMode(): boolean {
     return this.getArgument(ECliArgument.testMode) as boolean
   }
 
+  /**
+   * Checks if an alias for the main NPM package file should be
+   * added to the generated .d.ts source
+   */
+  private noAlias(): boolean {
+    return this.getArgument(ECliArgument.noAlias) as boolean
+  }
+
   /**
    * Checks if script is forced to attempt generation despite errors
    */
   private useForce(): boolean {
     return this.getArgument(ECliArgument.force) as boolean
   }
 
+  /**
+   * Checks how and if tree-shaking should be applied
+   */
+  private getShake(): 'off' | 'exportOnly' | 'allImports' {
+    const shake = this.getArgument(ECliArgument.shake) as
+      | undefined
+      | 'off'
+      | 'exportOnly'
+      | 'allImports'
+    if (shake === undefined) return 'off'
+    if (shake === 'exportOnly' || shake === 'allImports' || shake === 'off')
```

Comment:
When enum is used, validation could be implemented like this:

If you would use enum, you could test like this:

```
if (Object.values(ShakeOptions).includes(shake)) {
   return shake
} else {
....
```